tests/test_lightningd.py

```diff
@@ -106,6 +107,9 @@ def get_node(self, disconnect=None, options=None, may_fail=False, random_hsm=Fal
         lightning_dir = os.path.join(
             TEST_DIR, self.testname, "lightning-{}/".format(node_id))
 
+        if os.path.exists(lightning_dir):
```

Comment:
I thought about that as well, but `setUp` and `tearDown` are run for failing tests as well, so we'd be removing state that we may want to inspect (I often open the DB with `sqlite3` on failing runs and look what could have gone wrong for example).

`setUp` is also problematic since we don't know the number of `lightningd`s we are going to spawn, so we'd be left with the option of removing the tests parent directory which contains all the daemon directories. It's an option, but this is a bit more targeted.

Reply:
Ah, I thought `tearDown` already performed deletion of DBs somehow.  It turns out we allocate a new subdirectory for each test case.  So I suppose this is the best we can do.

Reply:
Yeah, I like leaving things on disk for the eventuality that we need to go in and inspect things :smiley: